packages/ai/core/generate-text/generate-text.ts

```diff
@@ -274,15 +275,17 @@ functionality that can be fully encapsulated in the provider.
             }),
             tracer,
             fn: async span => {
+              const { signal: effectiveAbortSignal, clearTimeout } = getEffectiveAbortSignal(abortSignal, timeout);
               const result = await model.doGenerate({
                 mode,
                 ...callSettings,
                 inputFormat: currentInputFormat,
                 prompt: promptMessages,
                 providerMetadata,
-                abortSignal,
+                abortSignal: effectiveAbortSignal, 
                 headers,
               });
```

Comment:
needs to happen in a finally block in case of errors

Reply:
fixed